packages/react-native/Libraries/Components/TextInput/TextInput.js

```diff
@@ -1206,6 +1206,24 @@ function InternalTextInput(props: Props): React.Node {
     }
   }, [inputRef]);
 
+  useLayoutEffect(() => {
+    if (inputRef.current != null) {
+      viewCommands.setTextAndSelection(
```

Comment:
With this effect, `viewCommands.setTextAndSelection` is going to be invoked whenever any of the dependencies (e.g. `lastNativeSelection`, etc.) change — not only when `props.secureTextEntry` changes.

I think you need a ref that tracks the value of `props.secureTextEntry` the last time that this effect was invoked, and only invoke this if that prop has changed.

Reply:
Good point @yungsters !

I'm going to apply your suggestion 😉

Reply:
@yungsters 

It is done 😄